src/wp-includes/media.php

```diff
@@ -5578,7 +5578,45 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 	 * It is here to avoid duplicate logic in many places below, without having
 	 * to introduce a very specific private global function.
 	 */
-	$postprocess = static function( $loading_attributes, $with_fetchpriority = false ) use ( $tag_name, $attr, $context ) {
+	$postprocess = static function( $loading_attributes, $with_fetchpriority = false, $is_header = false ) use ( $tag_name, $attr, $context ) {
+		if ( $is_header ) {
+			/** This filter is documented in wp-admin/includes/media.php */
+			$wp_min_priority_img_pixels = apply_filters( 'wp_min_priority_img_pixels', 50000 );
+			// Images with a certain minimum size in the header of the page are also counted towards the threshold.
+			if ( $wp_min_priority_img_pixels <= $attr['width'] * $attr['height'] ) {
+				wp_increase_content_media_count();
+			}
+		}
+		// Potentially add `fetchpriority="high"`.
+		if ( $with_fetchpriority ) {
+			$loading_attributes = wp_maybe_add_fetchpriority_high_attr( $loading_attributes, $tag_name, $attr );
+		}
+		// Potentially strip `loading="lazy"` if the feature is disabled.
+		if ( isset( $loading_attributes['loading'] ) && ! wp_lazy_loading_enabled( $tag_name, $context ) ) {
+			unset( $loading_attributes['loading'] );
+		}
+		return $loading_attributes;
+	};
+	/*
+	 * Closure for postprocessing logic.
+	 * It is here to avoid duplicate logic in many places below, without having
+	 * to introduce a very specific private global function.
+	 */
+	$postprocess = static function( $loading_attributes, $with_fetchpriority = false, $is_header = false ) use ( $tag_name, $attr, $context ) {
```

Comment:
I am confused, why are you overriding the closure here?

I am starting to feel like this closure doesn't make any sense and should be a function in it's own right. A closure if useful to stop a couple of lines from being repeated. This is getting more and more complex. It should have it's own unit tests and it is hard to test a closure. Should / could be move it's own function and add unit tests to it.

Reply:
I added the new code here as it was repeating at a couple of places where this closure was called. Since the objective of the closure was mainly to avoid repetition, I modified the same closure by adding a new parameter and a few lines to it. 

After giving it a thought, I feel, you are correct @spacedmonkey; this has turned out to be more confusing. I did better separate it out to a new closure. Own function isn't the right option as both the lines inside the closure are not so related, and these lines will only be ever required by the current parent.

Reply:
@kt-12 Both of these closures are assigned to the variable `$postprocess`. The second one would overwrite the first one. Making the first one unused and pointless. What is going on here?

Reply:
+1, @kt-12 please check, the code looks like it's exactly the same. I'm not sure why the closure is duplicated now, it wasn't like that when I reviewed yesterday.

If you want to implement to closures, then one should remain the original `$postprocess` closure as is, and another one should be _only_ for the logic in the `$is_header` if clause, this is how you can break them apart in a clean way. But right now you're simply duplicating the same closure.

Reply:
@spacedmonkey @felixarntz  this was an accidental commit, I have fixed it now. Also, I have moved the counting code to a separate closure.